boards/common/particle-mesh/board.c

```diff
@@ -89,3 +89,11 @@ void board_init(void)
     /* initialize the CPU */
     cpu_init();
 }
+
+bool board_bootloader_alternative_mode(void)
+{
+    /* If the application button is pressed, this indicates that the bootloader
+     * should switch to its alternative mode (eg. DFU recovery). */
+    gpio_init(BTN0_PIN, BTN0_MODE);
+    return !gpio_read(BTN0_PIN);
+}
```

Comment:
The button should probably be configurable no? I imagine many will implement this, and it will pretty much be the same only maybe with different pin numbers.

Reply:
If we're going with the Ben's suggestion, the `board.h` of providing boards would
```c
#eefine BTN_BOOTLOADER_PIN BTN0_PIN
```
etc -- but per board it can IMO be fixed (especially on boards that have only one button)